utils/lib.sh

```diff
@@ -1,104 +1,119 @@
-# Library of functions for Calico process and release automation.
+# Library of functions for Felix process and release automation.
 
 # Get the root directory of the Git repository that we are in.
 function git_repo_root {
     git rev-parse --show-toplevel
 }
 
-# Update the RPM spec file for the release.  Environment required: ${package},
-# ${nextrel}.
-function update_rpm_spec {
-
-    origd=`pwd`
-    cd `git_repo_root`/rpm
+# Get the current Git branch.
+function git_current_branch {
+    git rev-parse --abbrev-ref HEAD
+}
 
-    rpm_spec=${package}.spec
+# Get the last tag.
+function git_last_tag {
+    git describe --tags --abbrev=0
+}
 
-    # Generate RPM version and release.
-    IFS=- read version qualifier <<< ${nextrel}
-    if test "${qualifier}"; then
-	rpmver=${version}
-	rpmrel=0.1.${qualifier}
+# Autogenerate PEP 440 version based on current Git state.
+function git_auto_version {
```

Comment:
Mostly I think just because we still have some Python code here (in Felix) for the moment - so still want a PEP 440 version.

There is another small reason, though: if you're working iteratively with uncommitted code, `git describe` on its own doesn't give you a monotonically increasing version.